server/mediapublic/mediapublic/views.py

```diff
@@ -0,0 +1,686 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+
+from sqlalchemy.exc import DBAPIError
+
+from .models import (
+    DBSession,
+    Users,
+    UserTypes,
+    RecordingCategories,
+    Comments,
+    Organizations,
+    People,
+    Recordings,
+    Howtos,
+    Blogs,
+    Playlists,
+    PlaylistAssignments,
+    )
+
+import datetime
+import json
+
+def check_payload(body, cls, keys=[]):
+
+    dt_objs = [
+        'creation_datetime',
+        'edit_datetime',
+        'recorded_datetime',
+    ]
+
+    payload = {}
+    resp = None
+    #code = 200
+    try:
+        payload = json.loads(body)
+        if not isinstance(payload, dict):
+            payload = {}
+            resp = {'error': 'JSON not dictionary.'}
+            #code = 400
+        else:
+            reqkeys = cls.reqkeys()
+            if not ( (len(reqkeys) + len(keys)) == len(payload) and \
+                    all(k in payload for k in reqkeys) and all(k in payload for k in keys) ):
+                payload = {}
+                resp = {'error': 'Missing Key.  Required Keys: %s, %s' % (cls.reqkeys(), keys) }
+                #code = 400
+            else:
+                for k in payload:
+                    if k in dt_objs:
+                        try:
+                            payload[k] = datetime.datetime.strptime(payload[k], "%Y-%m-%d %H:%M:%S.%f")
+                        except:
+                            payload = {}
+                            resp = {'error': 'Invalid DateTime format for key: %s' % k}
+                            break
+    except:
+        resp = {'error': 'Invalid JSON'}
+    #    #code = 400
+
+    return payload, resp #, code
+
+def do_post(request, keys, cls,):
+    resp = {'error': 'Invalid Input'}
+    status = 400
+    payload, err = check_payload(request.body, cls)
+    if err is None:
+        for k in keys:
+            payload[k] = keys[k]
+        payload['creation_datetime'] = datetime.datetime.now()
```

Comment:
Do you have a preferred way of doing validation? Seems like there should be a hook in here somewhere (and in do_put)